pandas/core/internals.py

```diff
@@ -4802,6 +4802,11 @@ def get_reindexed_values(self, empty_dtype, upcasted_na):
 
             if self.is_null and not getattr(self.block, 'is_categorical',
                                             None):
+                if isinstance(empty_dtype, DatetimeTZDtype):
+                    # handle timezone-aware all NaT cases
+                    return DatetimeIndex([fill_value] * np.prod(self.shape),
+                                         dtype=empty_dtype)
+
```

Comment:
I think we need a method on ``Block`` like ``empty(self, shape)`` to create an empty block filled with the missing value for that type.